Wrap Selections in both Quicktags Editor and TinyMCE w/ Shortcodes

JayWood opened the pull request:
Speaks for itself, this will wrap the user's selection in the short code tag, complete w/ closing tags.

JayWood committed:
Allow shortcodes to work with selected text

JayWood committed:
Update version

jtsternberg commented on js/shortcode-quicktag-button.js:
```diff
@@ -111,6 +113,24 @@ window.wp_sc_buttons.qt = (function(window, document, $, QTags, buttons, scbutto
+					startPos = canvas.selectionStart,
+					endPos = canvas.selectionEnd;
+
+				window.console.log( startPos,endPos );
```
is this log necessary? Also, if you're logging, use the logging method, `scbuttons.log`.

jtsternberg commented on js/shortcode-quicktag-button.js:
```diff
@@ -103,7 +103,9 @@ window.wp_sc_buttons.qt = (function(window, document, $, QTags, buttons, scbutto
+			var shortcode = '['+ params.slug,
+				close_tag = '[/' + params.slug + ']',
+				mce_selection = ( tinymce.activeEditor ) ? tinymce.activeEditor.selection.getContent() : false;
 
```
Please, no inline var declaration. One var statement/line per var declaration. ([reference](http://benalman.com/news/2012/05/multiple-var-statements-javascript/))

jtsternberg commented on js/shortcode-quicktag-button.js:
```diff
@@ -103,7 +103,9 @@ window.wp_sc_buttons.qt = (function(window, document, $, QTags, buttons, scbutto
-			var shortcode = '['+ params.slug;
+			var shortcode = '['+ params.slug,
+				close_tag = '[/' + params.slug + ']',
+				mce_selection = ( tinymce.activeEditor ) ? tinymce.activeEditor.selection.getContent() : false;
```
Does this work on/in the text view? I know there's a way to get the selection in the text view as well.

jtsternberg commented:
I prefer that the close tag not be used if there is no selection. (i.e. use the self-closing shortcode)

jtsternberg commented:
This should be a parameter that is registered when `_Shortcode_Button_` is first initiated, whether shortcode can wrap existing text. if not, should use default (existing) handling.

JayWood committed:
Remove rogue log, no inline var declaration

JayWood commented:
The log was just an oopsie on my part, but i'm sad that inline var declarations aren't a thing, they look so much cleaner.

I'll see what I can do about the registration parameter for closing tags.  But that begs the question, if the user says 'yes' to closing tags, ie, the shortcode must use a selection, then wouldn't we require the user to make a selection before they can even use the shortcode?

JayWood commented:
Oh, and yes this currently works with the text only mode as well.

jtsternberg commented:
By 'user', I think you mean the dev who initiates, right? And no, i wasn't saying a selection should be required, though the parameter could take 'required' as a value if you want to make your PR work that way. the parameter would default to false (not allowed), could toggle to true (allowed) or 'required'.

JayWood committed:
Add include_close in registration params, better QT handling

JayWood committed:
Update readme to reflect new paramater

JayWood committed:
Recompile JS

JayWood committed:
Actually force close tag ( see comments )

If force close is set, this will force a closing tag, regardless
of user's selection.  However, if user has a selection, it will be
of course included and wrapped.

jtsternberg commented:
Thanks dude!
